### PR TITLE
Nginx Instrumentation crd

### PR DESCRIPTION
**Pull request by chrlic**
Instrumentation crd for Nginx autoinstrumentation

Aside of standard fields, it contains two more:
- Attrs - allows specific settings for instrumentation library, comment points to the doc and will go into README.md eventually
- ConfigFile - allows to specify custom configuration file location for Nginx if it's different from most used /etc/nginx/nginx.conf. This is needed since the instrumentation essentially modifies this file to load the instrumentation shared library and to set OpenTelemetry parameters.

In a nutshell, Nginx instrumentation follows similar methods to Apache HTTPD instrumentation.

### CODE REVIEW

File: apis/v1alpha1/instrumentation_types.go [hunk 1]
```diff
@@ -218,6 +222,35 @@ type ApacheHttpd struct {
...
+	// Resources describes the compute resource requirements.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resourceRequirements,omitempty"`
+}
+
+// Nginx defines Nginx SDK and instrumentation configuration.
+type Nginx struct {
+	// Image is a container image with Nginx SDK and auto-instrumentation.
+	// +optional
+	Image string `json:"image,omitempty"`
```
Comment (pavolloffay) [inline apis/v1alpha1/instrumentation_types.go:235]: Is there already image for nginx instrumentation?

**Review by pavolloffay — COMMENTED** [review]
Cool to see more auto-instrumentation support :) 

Could you please share the link where the nginx instrumentation is hosted?

File: apis/v1alpha1/instrumentation_types.go (hunk 1 above)
Comment (pavolloffay) [reply apis/v1alpha1/instrumentation_types.go:235]: ah it's here https://github.com/open-telemetry/opentelemetry-operator/pull/1852

Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:235]: Here's the Nginx Otel module: https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/main/instrumentation/nginx

Comment (chrlic) [reply apis/v1alpha1/instrumentation_types.go:235]: For Nginx autoinstrumentation, default image with the instrumentation libraries is the same as for Apache Httpd - autoinstrumentation-apache-httpd

**Comment by pavolloffay** [comment]
The PR needs rebasing

File: apis/v1alpha1/instrumentation_webhook.go [hunk 2]
```diff
@@ -170,6 +171,26 @@ func (r *Instrumentation) Default() {
...
 		r.Spec.ApacheHttpd.ConfigPath = "/usr/local/apache2/conf"
 	}
+	if r.Spec.Nginx.Image == "" {
+		if val, ok := r.Annotations[AnnotationDefaultAutoInstrumentationNginx]; ok {
+			r.Spec.Nginx.Image = val
+		}
+	}
+	if r.Spec.Nginx.Resources.Limits == nil {
+		r.Spec.Nginx.Resources.Limits = corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("500m"),
```
Comment (pavolloffay) [inline apis/v1alpha1/instrumentation_webhook.go:181]: we should extract these to a constant and reuse for others.

**Review by pavolloffay — APPROVED** [review]

File: apis/v1alpha1/instrumentation_webhook.go (hunk 2 above)
Comment (chrlic) [reply apis/v1alpha1/instrumentation_webhook.go]: done

**Comment by chrlic** [comment]
rebased, reopenning

**Review by pavolloffay — APPROVED** [review]
